aci/resource_aci_cloudtemplateextnetwork_test.go

```diff
@@ -0,0 +1,149 @@
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckAciCloudTemplateforExternalNetworkConfig_basic(fv_tenant_name, cloudtemplate_infra_network_name, cloudtemplate_ext_network_name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAciCloudTemplateforExternalNetworkExists("aci_templatefor_external_network.footemplatefor_external_network", &templatefor_external_network),
+					testAccCheckAciCloudTemplateforExternalNetworkAttributes(fv_tenant_name, cloudtemplate_infra_network_name, cloudtemplate_ext_network_name, description, &templatefor_external_network),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAciCloudTemplateforExternalNetwork_Update(t *testing.T) {
+	var templatefor_external_network models.CloudTemplateforExternalNetwork
+	fv_tenant_name := acctest.RandString(5)
+	cloudtemplate_infra_network_name := acctest.RandString(5)
+	cloudtemplate_ext_network_name := acctest.RandString(5)
+	description := "templatefor_external_network created while acceptance testing"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAciCloudTemplateforExternalNetworkDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckAciCloudTemplateforExternalNetworkConfig_basic(fv_tenant_name, cloudtemplate_infra_network_name, cloudtemplate_ext_network_name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAciCloudTemplateforExternalNetworkExists("aci_templatefor_external_network.footemplatefor_external_network", &templatefor_external_network),
+					testAccCheckAciCloudTemplateforExternalNetworkAttributes(fv_tenant_name, cloudtemplate_infra_network_name, cloudtemplate_ext_network_name, description, &templatefor_external_network),
+				),
+			},
+			{
+				Config: testAccCheckAciCloudTemplateforExternalNetworkConfig_basic(fv_tenant_name, cloudtemplate_infra_network_name, cloudtemplate_ext_network_name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAciCloudTemplateforExternalNetworkExists("aci_templatefor_external_network.footemplatefor_external_network", &templatefor_external_network),
+					testAccCheckAciCloudTemplateforExternalNetworkAttributes(fv_tenant_name, cloudtemplate_infra_network_name, cloudtemplate_ext_network_name, description, &templatefor_external_network),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAciCloudTemplateforExternalNetworkConfig_basic(fv_tenant_name, cloudtemplate_infra_network_name, cloudtemplate_ext_network_name string) string {
+	return fmt.Sprintf(`
+
+	resource "aci_tenant" "footenant" {
+		name 		= "%s"
+		description = "tenant created while acceptance testing"
+
+	}
+
+	resource "aci_infra_network_template" "fooinfra_network_template" {
+		name 		= "%s"
+		description = "infra_network_template created while acceptance testing"
+		tenant_dn = aci_tenant.footenant.id
+	}
+
+	resource "aci_templatefor_external_network" "footemplatefor_external_network" {
```

Comment:
aci_templatefor_external_network - is not a valid resource and please update all other references.